### DISCUSSION TITLE
Prepopulate image with existing world data

### DISCUSSION THREAD

**Discussion by giantcow** [lloesche/valheim-server-docker#559] [Q&A] [answered]
I come from the AWS ECS package (https://github.com/rileydakota/valheim-ecs-fargate-cdk), and I am working on allowing users to populate the docker image with a world that already exists. https://github.com/RyoshiKayo/valheim-ecs-fargate-cdk/blob/bootstrap-world/lib/valheim-server-aws-cdk-stack.ts#L98-L99

Questions:
- What is the "correct" hook to populate the world data folder?
- Where should I be copying this world data too?
  - Tried `/home/valheim/.config/unity3d/IronGate/Valheim/worlds_local/` and `/config/unity3d/IronGate/Valheim/worlds_local/`

**Comment by wcdanh2** [comment]
Check the "basic docker usage" section of the Readme. You don't really need a hook you just mount a folder that contains your worlds_local folder with worlds inside.

**Reply by giantcow** [reply]
In the ECS package, we're deploying the drive we mount to and the container at basically the same time. It would be a lot more cumbersome to create a new container/host only to upload the world files when we could just utilize the hooks to do it for us.

**Comment by giantcow** [comment, marked as answer]
Figured this out. In the ECS repo we are mounting all of `/config` to a Elastic File System file system. I was running into issues because the backup I took from an older Valheim server version had a different format than what this docker image was using. So I had to [reformat the ZIP archive](https://github.com/RyoshiKayo/valheim-ecs-fargate-cdk/blob/bootstrap-world/README.md?plain=1#L61-L91) and [change the destination folder](https://github.com/RyoshiKayo/valheim-ecs-fargate-cdk/blob/bootstrap-world/lib/valheim-server-aws-cdk-stack.ts#L101-L102)

I was referencing files in `/home/valheim/` because some of the log messages referenced those locations, even though it looks like we're not directly using them..
